Libraries/LibGfx/Bitmap.cpp

```diff
@@ -182,6 +182,23 @@ Bitmap::Bitmap(BitmapFormat format, NonnullRefPtr<SharedBuffer>&& shared_buffer,
         allocate_palette_from_format(m_format, palette);
 }
 
+RefPtr<Gfx::Bitmap> Bitmap::clone() const
+{
+    const auto w = this->width();
+    const auto h = this->height();
+    const auto purgeable = this->m_purgeable ? Purgeable::Yes : Purgeable::No;
+
+    auto new_bitmap = new Bitmap(this->format(), { w, h }, purgeable);
+    if (!new_bitmap) {
+        return nullptr;
+    }
```

Comment:
`new` itself will not fail here since it's only concerned with heap-allocating `sizeof(Bitmap)`
The real issue is the size of the backing store for the bitmap.
We should reorganize `Gfx::Bitmap` construction so that the backing store is allocated before we create and return a `Gfx::Bitmap`.

Reply:
That makes sense. I've restructured the above to use the static `create()` methods, so as to as least limit where `new`s are happening.

Also had a bash at your reorganisation suggestion by moving backing store allocation to the appropriate static `create()` methods. I've introduced a `BackingStore` convenience struct so that we don't have to risk recalculating the pitch in several different places.

I'm still a little nervous that the size of the allocated memory chunk is not coupled to the backing store at all, as the latter is ultimately just a `void*`. That said, the constructor is private so maybe it's not a big deal.